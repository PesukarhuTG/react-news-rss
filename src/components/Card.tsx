import React from 'react';
import styled from 'styled-components';
import CardProps from '../types/Card';

class Card extends React.Component<CardProps> {
  constructor(props: CardProps) {
    super(props);
  }

  render() {
    const { author, description, publishedAt, title, urlToImage } = this.props;

    return (
      <Item data-testid="card-item">
        <CardImage
          style={{
            backgroundImage: `url(${urlToImage || '../assets/img/no-poster.jpg'}`,
          }}
        />
        <Title>{title}</Title>
        <Description>{description || 'Sorry, there is no any description'}</Description>
        <NewsDate>Data: {publishedAt.slice(0, 10)}</NewsDate>
        <Author>Author: {author || 'unnamed'}</Author>
      </Item>
    );
  }
}

const Item = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: 10px;
  position: relative;
  padding-bottom: 10px;
  width: 100%;
  overflow: hidden;
  background-color: var(--second-contrast);
  border-radius: 10px;
  list-style: none;
  transition: 0.3s all;

  &:hover {
    box-shadow: 0 0 15px 0 var(--primary);
  }
`;

const Title = styled.p`
  max-height: 50px;
  height: 100%;
  font-size: 18px;
  color: var(--primary);
  margin: 0 10px;
  text-transform: uppercase;

  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
`;

const Description = styled.p`
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
  overflow: hidden;

  max-height: 90px;
  height: 100%;
  margin: 0 10px;
  font-size: 12px;
`;

const NewsDate = styled.p`
  display: block;
  margin: 0 10px;
  font-size: 12px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  padding-top: 20px;
  border-top: 1px dotted var(--main-color);
`;

const Author = styled.p`
  display: -webkit-box;
  margin: 0 10px;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  overflow: hidden;
  font-size: 12px;
`;

const CardImage = styled.div`
  display: block;
  width: 100%;
  height: 265px;
  background-image: url('../assets/img/no-poster.jpg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  border-radius: 10px 10px 0 0;
`;

export default Card;
