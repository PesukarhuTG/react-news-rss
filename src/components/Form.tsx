import React from 'react';
import FormProps from '../types/Form';

interface LoginFormProps {
  onSubmit: (data: FormProps) => void;
}

interface FormFields {
  name: HTMLInputElement;
  birthday: HTMLInputElement;
  city: HTMLInputElement;
  gender: HTMLInputElement;
  file: HTMLInputElement;
  remember: HTMLInputElement;
}

interface State {
  disabled: boolean;
  selectedFile: any;
}

class LoginForm extends React.Component<LoginFormProps, State> {
  constructor(props: LoginFormProps) {
    super(props);
    this.state = {
      disabled: true,
      selectedFile: null,
    };
  }

  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (e.target.files !== null) {
      this.setState({ selectedFile: e.target.files[0] });
    }
  };

  handleSubmit: React.FormEventHandler<HTMLFormElement & FormFields> = (event) => {
    event.preventDefault();
    const form = event.currentTarget;
    const { name, birthday, city, gender, file, remember } = form;

    const blob = new Blob([this.state.selectedFile]);
    const url = URL.createObjectURL(blob);

    this.props.onSubmit({
      name: name.value,
      birthday: birthday.value,
      city: city.value,
      gender: gender.value,
      file: url,
      remember: remember.checked,
    });

    this.clearForm([name, birthday, city, gender, file, remember]);
  };

  clearForm = (prop: Array<HTMLInputElement>) => {
    prop.forEach((item) => {
      if (item.type === 'checkbox') {
        item.checked = false;
      } else {
        item.value = '';
      }
    });

    this.setState({ disabled: true });
    this.setState({ selectedFile: null });
  };

  setDisabled = (value: string) => {
    value.length !== 0 ? this.setState({ disabled: false }) : this.setState({ disabled: true });
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} data-testid="form">
        <div className="form-wrapper">
          <label className="input-wrapper">
            <span className="input-name">Name</span>
            <input
              className="input-field"
              name="name"
              type="text"
              data-testid="input-fname"
              onChange={(e) => this.setDisabled(e.target.value)}
              required
            />
          </label>
          <label className="input-wrapper">
            <span className="input-name">Birthday</span>
            <input
              className="input-field"
              name="birthday"
              type="date"
              data-testid="input-fdata"
              onChange={(e) => this.setDisabled(e.target.value)}
              required
            />
          </label>
        </div>

        <div className="form-wrapper">
          <label className="input-wrapper">
            <span className="input-name">Choose a city</span>
            <select
              className="input-city"
              name="city"
              data-testid="input-fcity"
              onChange={(e) => this.setDisabled(e.target.value)}
              required
            >
              <option value="" />
              <option value="Saint-Petersburg">Saint-Petersburg</option>
              <option value="Moscow">Moscow</option>
              <option value="Voronezh">Voronezh</option>
              <option value="Petrozavodsk">Petrozavodsk</option>
              <option value="other city">...other city</option>
            </select>
          </label>
          <label className="input-wrapper">
            <span className="input-name">Gender</span>
            <div className="genders-wrapper">
              <input
                type="radio"
                id="man"
                name="gender"
                value="man"
                data-testid="input-fgender-man"
                defaultChecked
              />
              <label className="gender-name" htmlFor="man">
                man
              </label>

              <input
                type="radio"
                id="woman"
                name="gender"
                value="woman"
                data-testid="input-fgender-woman"
              />
              <label className="gender-name" htmlFor="woman">
                woman
              </label>
            </div>
          </label>
        </div>

        <label className="input-wrapper">
          <span className="input-name">Upload a file</span>
          <input
            className="input-file"
            name="file"
            type="file"
            accept="image/*, .png, .jpg"
            data-testid="input-file"
            onChange={this.handleChange}
          />
        </label>

        <label className="input-wrapper-gorizont">
          <input name="remember" type="checkbox" data-testid="input-faccept" />
          <span className="remember-name">I consent to use my personal data</span>
        </label>

        <button
          className="btn-submit"
          type="submit"
          disabled={this.state.disabled}
          data-testid="btn-submit"
        >
          Send my data
        </button>
      </form>
    );
  }
}

export default LoginForm;
