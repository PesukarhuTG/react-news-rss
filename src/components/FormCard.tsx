import React from 'react';
import styled from 'styled-components';
import FormProps from '../types/Form';

class FormCard extends React.Component<FormProps> {
  constructor(props: FormProps) {
    super(props);
  }

  render() {
    const { name, birthday, city, gender, file } = this.props;

    return (
      <Item data-testid="form-card">
        <CardImage
          style={{
            backgroundImage: `url(${file})`,
          }}
        />
        <Description>
          <Name>{name}</Name>
          <BirthdayData>
            <strong>Birthday: </strong>
            {birthday}
          </BirthdayData>
          <GenderData>
            <strong>Gender: </strong>
            <span style={{ color: `${gender === 'man' ? 'var(--primary)' : 'var(--gender-w)'}` }}>
              {gender}
            </span>
          </GenderData>
          <CityData>
            <strong>City: </strong>
            {city}
          </CityData>
        </Description>
      </Item>
    );
  }
}

const Item = styled.li`
  display: flex;
  flex-direction: column;
  gap: 10px;
  padding: 10px;
  background-color: var(--second-contrast);
  border-radius: 10px;
  list-style: none;
  transition: 0.3s all;

  &:hover {
    box-shadow: 0 0 15px 0 var(--primary);
  }
`;

const Description = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
`;

const Name = styled.p`
  font-size: 18px;
  font-weight: 700;
  color: var(--primary);
  text-transform: uppercase;
`;

const BirthdayData = styled.span`
  font-size: 14px;
`;

const GenderData = styled.span`
  font-size: 14px;
`;

const CityData = styled.p`
  font-size: 14px;
`;

const CardImage = styled.div`
  display: block;
  width: 100px;
  height: 100px;
  margin: 0 auto;
  background-color: var(--avatar-background);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  border-radius: 50%;
`;

export default FormCard;
