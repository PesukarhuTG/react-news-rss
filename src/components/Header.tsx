import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

class Header extends React.Component {
  activeClassName = 'active-link';

  render() {
    return (
      <AppHeader>
        <NavLink to="/" className={'logo'}>
          RACOON digest
        </NavLink>

        <HeaderNav>
          <NavLink
            to="/"
            className={({ isActive }) => (isActive ? this.activeClassName : undefined)}
            data-testid="mainpage-link"
            end
          >
            Home
          </NavLink>
          <NavLink
            to="/about"
            className={({ isActive }) => (isActive ? this.activeClassName : undefined)}
            data-testid="aboutpage-link"
          >
            About
          </NavLink>
          <NavLink
            to="/contacts"
            className={({ isActive }) => (isActive ? this.activeClassName : undefined)}
            data-testid="contactspage-link"
          >
            Contacts
          </NavLink>
        </HeaderNav>
      </AppHeader>
    );
  }
}

const AppHeader = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 0;
`;

const HeaderNav = styled.nav`
  display: flex;
  margin-left: auto;
  gap: 30px;
`;

export default Header;
