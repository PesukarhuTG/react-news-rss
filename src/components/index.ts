export { default as Layout } from './Layout';
export { default as Header } from './Header';
export { default as SearchPanel } from './Search';
export { default as CardsAlbum } from './CardsAlbum';
export { default as Card } from './Card';
export { default as LoginForm } from './Form';
export { default as FormCard } from './FormCard';
export { default as FormCardsAlbum } from './FormCardsAlbum';
export { default as Spinner } from './Spinner';
export { default as Modal } from './Modal';
