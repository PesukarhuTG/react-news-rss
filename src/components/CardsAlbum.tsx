import React from 'react';
import styled from 'styled-components';
import CardProps from '../types/Card';
import Card from './Card';

interface CardListProps {
  cards?: CardProps[];
}

class CardsAlbum extends React.Component<CardListProps> {
  constructor(props: CardListProps) {
    super(props);
  }

  render() {
    if (!this.props.cards?.length) {
      return <LoadMessage>Data is preparing...</LoadMessage>;
    }

    const { cards } = this.props;
    cards.length = 20;

    return (
      <Album>
        {cards.map((item: CardProps, index: number) => {
          return <Card {...item} key={index} />;
        })}
      </Album>
    );
  }
}

const Album = styled.ul`
  margin: 0 auto;
  width: 100%;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  justify-content: space-between;
  gap: 20px;
  margin-bottom: 50px;
`;

const LoadMessage = styled.p`
  text-align: center;
  padding: 20px 0;
`;

export default CardsAlbum;
