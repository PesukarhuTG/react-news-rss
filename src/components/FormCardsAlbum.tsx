import React from 'react';
import styled from 'styled-components';
import FormProps from '../types/Form';
import FormCard from './FormCard';

interface CardListProps {
  list?: FormProps[];
}

class FormCardsAlbum extends React.Component<CardListProps> {
  constructor(props: CardListProps) {
    super(props);
  }

  render() {
    if (!this.props.list?.length) {
      return;
    }

    const { list } = this.props;

    return (
      <Album>
        {list.map((item: FormProps, index: number) => {
          return <FormCard {...item} key={index} />;
        })}
      </Album>
    );
  }
}

const Album = styled.ul`
  margin: 20px auto;
  width: 100%;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  justify-content: space-between;
  gap: 20px;
  margin-bottom: 50px;
`;

export default FormCardsAlbum;
