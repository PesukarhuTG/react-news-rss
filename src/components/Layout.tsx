import React from 'react';
import Header from './Header';
import styled from 'styled-components';

interface LayoutProps {
  children?: React.ReactNode;
}

class Layout extends React.Component<LayoutProps> {
  render() {
    return (
      <Container data-testid="layout-container">
        {<Header />}
        {this.props.children}
      </Container>
    );
  }
}

const Container = styled.div`
  max-width: 1440px;
  width: 100%;
  min-height: 100vh;
  margin: 0 auto;
  padding-left: 20px;
  padding-right: 20px;
  display: flex;
  flex-direction: column;
`;

export default Layout;
