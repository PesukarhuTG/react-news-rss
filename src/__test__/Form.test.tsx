import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LoginForm } from '../components';

describe('Form tests', () => {
  test('render Form', () => {
    render(<LoginForm onSubmit={() => {}} />);
    expect(screen.getByTestId('form')).toBeInTheDocument();
  });

  test('render Form inputs on page', () => {
    render(<LoginForm onSubmit={() => {}} />);
    expect(screen.getByTestId('input-file')).toBeInTheDocument();
  });

  test('Form snapshot', () => {
    const form = render(<LoginForm onSubmit={() => {}} />);
    expect(form).toMatchSnapshot();
  });

  test('input name: check form value', () => {
    render(<LoginForm onSubmit={() => {}} />);

    const input = screen.getByTestId('input-fname');
    expect(input).toBeInTheDocument();
    expect(input).toContainHTML('');
    fireEvent.input(input, {
      target: { value: 'Tany' },
    });
    expect(input).toHaveValue('Tany');
  });

  test('input radio: check render', () => {
    render(<LoginForm onSubmit={() => {}} />);

    const input = screen.getAllByRole('radio') as HTMLInputElement[];
    expect(input[0]).toBeInTheDocument();
    expect(input[1]).toBeInTheDocument();
    expect(input[0].value).toBe('man');
    expect(input[1].value).toBe('woman');
  });

  test('input data: check render', () => {
    render(<LoginForm onSubmit={() => {}} />);

    const input = screen.getByTestId('input-fdata');
    expect(input).toBeInTheDocument();
    fireEvent.input(input, {
      target: { value: '2022-01-01' },
    });
    expect(input).toHaveValue('2022-01-01');
  });

  test('Upload Files', async () => {
    render(<LoginForm onSubmit={() => {}} />);

    const fakeFile = new File(['hello'], 'hello.png', { type: 'image/png' });
    const inputFile = screen.getByTestId('input-file') as HTMLInputElement;

    expect(inputFile).toBeInTheDocument();

    await waitFor(() =>
      fireEvent.change(inputFile, {
        target: { files: [fakeFile] },
      })
    );

    if (inputFile.files instanceof FileList) {
      expect(inputFile.files[0]).toStrictEqual(fakeFile);
    }
  });

  test('input checkbox: check render', () => {
    render(<LoginForm onSubmit={() => {}} />);

    const checkbox = screen.getByTestId('input-faccept');
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
    fireEvent.click(checkbox);
    expect(checkbox).toBeChecked();
  });

  test('select: check render', () => {
    render(<LoginForm onSubmit={() => {}} />);

    const select = screen.getByTestId('input-fcity') as HTMLSelectElement;

    expect(select).toBeInTheDocument();
    expect(screen.getAllByRole('option').length).toBe(6);
    fireEvent.change(select, {
      target: { value: 'Saint-Petersburg' },
    });
    expect(screen.getByText('Saint-Petersburg')).toBeInTheDocument();
  });

  test('render submit button', () => {
    render(<LoginForm onSubmit={() => {}} />);
    const button = screen.getByTestId('btn-submit');
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('btn-submit');
    expect(button).toBeDisabled();
  });
});
