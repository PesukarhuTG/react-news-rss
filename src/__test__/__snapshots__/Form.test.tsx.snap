// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form tests Form snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        data-testid="form"
      >
        <div
          class="form-wrapper"
        >
          <label
            class="input-wrapper"
          >
            <span
              class="input-name"
            >
              Name
            </span>
            <input
              class="input-field"
              data-testid="input-fname"
              name="name"
              required=""
              type="text"
            />
          </label>
          <label
            class="input-wrapper"
          >
            <span
              class="input-name"
            >
              Birthday
            </span>
            <input
              class="input-field"
              data-testid="input-fdata"
              name="birthday"
              required=""
              type="date"
            />
          </label>
        </div>
        <div
          class="form-wrapper"
        >
          <label
            class="input-wrapper"
          >
            <span
              class="input-name"
            >
              Choose a city
            </span>
            <select
              class="input-city"
              data-testid="input-fcity"
              name="city"
              required=""
            >
              <option
                value=""
              />
              <option
                value="Saint-Petersburg"
              >
                Saint-Petersburg
              </option>
              <option
                value="Moscow"
              >
                Moscow
              </option>
              <option
                value="Voronezh"
              >
                Voronezh
              </option>
              <option
                value="Petrozavodsk"
              >
                Petrozavodsk
              </option>
              <option
                value="other city"
              >
                ...other city
              </option>
            </select>
          </label>
          <label
            class="input-wrapper"
          >
            <span
              class="input-name"
            >
              Gender
            </span>
            <div
              class="genders-wrapper"
            >
              <input
                checked=""
                data-testid="input-fgender-man"
                id="man"
                name="gender"
                type="radio"
                value="man"
              />
              <label
                class="gender-name"
                for="man"
              >
                man
              </label>
              <input
                data-testid="input-fgender-woman"
                id="woman"
                name="gender"
                type="radio"
                value="woman"
              />
              <label
                class="gender-name"
                for="woman"
              >
                woman
              </label>
            </div>
          </label>
        </div>
        <label
          class="input-wrapper"
        >
          <span
            class="input-name"
          >
            Upload a file
          </span>
          <input
            accept="image/*, .png, .jpg"
            class="input-file"
            data-testid="input-file"
            name="file"
            type="file"
          />
        </label>
        <label
          class="input-wrapper-gorizont"
        >
          <input
            data-testid="input-faccept"
            name="remember"
            type="checkbox"
          />
          <span
            class="remember-name"
          >
            I consent to use my personal data
          </span>
        </label>
        <button
          class="btn-submit"
          data-testid="btn-submit"
          disabled=""
          type="submit"
        >
          Send my data
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      data-testid="form"
    >
      <div
        class="form-wrapper"
      >
        <label
          class="input-wrapper"
        >
          <span
            class="input-name"
          >
            Name
          </span>
          <input
            class="input-field"
            data-testid="input-fname"
            name="name"
            required=""
            type="text"
          />
        </label>
        <label
          class="input-wrapper"
        >
          <span
            class="input-name"
          >
            Birthday
          </span>
          <input
            class="input-field"
            data-testid="input-fdata"
            name="birthday"
            required=""
            type="date"
          />
        </label>
      </div>
      <div
        class="form-wrapper"
      >
        <label
          class="input-wrapper"
        >
          <span
            class="input-name"
          >
            Choose a city
          </span>
          <select
            class="input-city"
            data-testid="input-fcity"
            name="city"
            required=""
          >
            <option
              value=""
            />
            <option
              value="Saint-Petersburg"
            >
              Saint-Petersburg
            </option>
            <option
              value="Moscow"
            >
              Moscow
            </option>
            <option
              value="Voronezh"
            >
              Voronezh
            </option>
            <option
              value="Petrozavodsk"
            >
              Petrozavodsk
            </option>
            <option
              value="other city"
            >
              ...other city
            </option>
          </select>
        </label>
        <label
          class="input-wrapper"
        >
          <span
            class="input-name"
          >
            Gender
          </span>
          <div
            class="genders-wrapper"
          >
            <input
              checked=""
              data-testid="input-fgender-man"
              id="man"
              name="gender"
              type="radio"
              value="man"
            />
            <label
              class="gender-name"
              for="man"
            >
              man
            </label>
            <input
              data-testid="input-fgender-woman"
              id="woman"
              name="gender"
              type="radio"
              value="woman"
            />
            <label
              class="gender-name"
              for="woman"
            >
              woman
            </label>
          </div>
        </label>
      </div>
      <label
        class="input-wrapper"
      >
        <span
          class="input-name"
        >
          Upload a file
        </span>
        <input
          accept="image/*, .png, .jpg"
          class="input-file"
          data-testid="input-file"
          name="file"
          type="file"
        />
      </label>
      <label
        class="input-wrapper-gorizont"
      >
        <input
          data-testid="input-faccept"
          name="remember"
          type="checkbox"
        />
        <span
          class="remember-name"
        >
          I consent to use my personal data
        </span>
      </label>
      <button
        class="btn-submit"
        data-testid="btn-submit"
        disabled=""
        type="submit"
      >
        Send my data
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
